Migrate combobox to wdio

nikvalor opened the pull request:
#### Please provide a link to the associated issue.

#### Please provide a brief summary of this pull request.

#### Please check whether the PR fulfills the following requirements

- [ ] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [ ] tests for the changes that have been done
- [ ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [ ] update `README.md`
- [ ] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ ] Documentation Examples
- [ ] Stackblitz works for all examples

nikvalor committed:
Merge remote-tracking branch 'origin/master' into migrate-combobox-to-wdio

# Conflicts:
#	package-lock.json
#	package.json

nikvalor committed:
Covered main functional cases for combobox component

nikvalor committed:
Covered scenarios related to combobox component

nikvalor committed:
Fixed typo

AntonOlkhovskyi commented on e2e/wdio/driver/wdio.ts:
```diff
@@ -36,19 +36,28 @@ export class Wdio {
         $$(selector)[index].setValue(value);
     };
 
+    setValueWithDelay(selector: string, value: string, delay: number = 100, index: number = 0, waitTime = this.defaultWaitTime): void {
```
if no input clear let's call it  addValueWithDelay

AntonOlkhovskyi commented on e2e/wdio/driver/wdio.ts:
```diff
@@ -57,28 +66,36 @@ export class Wdio {
+        return $(selector).waitForDisplayed({ timeout: waitTime, reverse: true });
     };
 
+    waitForNotFocused(selector: string, index: number = 0, waitTime = this.defaultWaitTime):void {
```
Maybe use waitUntil with isFocused check inside. This one isn't waiting as for me. If it is a generic check let's put it in the assertion helper. 
expect is not belongs to webDriver class.

AntonOlkhovskyi commented on e2e/wdio/platform/pages/combobox.po.ts:
```diff
@@ -0,0 +1,162 @@
+    getDropdownOptionsText(type: string): string[] {
+        let textArr: string[] = [];
+            this.expandDropdown(type);
+            const optionsLength = webDriver.getElementArrayLength(this.optionsArray);
```
`return webDriver.getTextArr(this.optionsArray, 0,-1)`
Perhaps need to pull last changes from feature/e2e-test-migrate-to-wdio

AntonOlkhovskyi commented on e2e/wdio/driver/wdio.ts:
```diff
@@ -36,19 +36,28 @@ export class Wdio {
 
     getText(selector: string, index: number = 0, waitTime = this.defaultWaitTime): string {
-        $$(selector)[index].waitForDisplayed({ timeout: waitTime });
+      //  $$(selector)[index].waitForDisplayed({ timeout: waitTime });
```
this is necessary?

nikvalor committed:
Merge branch 'feature/e2e-test-migrate-to-wdio' into migrate-combobox-to-wdio

# Conflicts:
#	e2e/wdio/driver/wdio.ts
#	e2e/wdio/helper/assertion-helper.ts
#	e2e/wdio/platform/pages/info-label.po.ts
#	e2e/wdio/platform/tests/panel.e2e-spec.ts
#	e2e/wdio/platform/tests/split-menu-button.e2e-spec.ts
#	package-lock.json
#	package.json

nikvalor committed:
PR changes

InnaAtanasova committed:
docs: re-order the Table columns in the Customizable Columns example (#4048)

fundamental-bot committed:
chore(release): version 0.25.0-rc.48 build 13298 [ci skip]

dimamarksman committed:
feat: (Platform) table phase 2 (#4028)

* feat: (Platform) Table phase 2

* feat: sorting

* fix: remove unnecessary Reset code & add sorting demo

* feat: base filtering

* feat: column freeze

* feat: borderless table

* feat: grouping

* feat: add i18n marks

* feat: datasource & dataprovider

* feat: search field

* feat: tests

* chore: filtering (not finished)

* view settings filters (in progress)

* sync with upstream

* add custom filter

* fix filters

* update sorting and grouping modals

* in progress

* fix table state events

* clean up table pipes

* add some tests

* add table unit tests

* table: align density in examples

* remove extra pipe option

* align table grid documentation

* table example: add "description" to filter by

* table: fix not-grouped-value selection

* table: add comments for some fields

* Update libs/platform/src/lib/components/table/table.component.ts

Co-authored-by: jmarkowski <jdrekmarkowski@gmail.com>

* table: default array/observable list data source

* table: resolve pr comments

* table: fixes after QA test

* commit to trigger travis

* fix circular deps

* commit to trigger travis

Co-authored-by: olshansky <olshansky0@gmail.com>
Co-authored-by: jmarkowski <jdrekmarkowski@gmail.com>

fundamental-bot committed:
chore(release): version 0.25.0-rc.49 build 13301 [ci skip]

droshev committed:
fix: remove deprecated component file-input (#4067)

BREAKING CHANGE:
removed deprecated file-input

fundamental-bot committed:
chore(release): version 0.25.0-rc.50 build 13312 [ci skip]

droshev committed:
fix: remove badge as a standalone component #3350 (#4068)

BREAKING CHANGE:
removed badge as a standalone component

fundamental-bot committed:
chore(release): version 0.25.0-rc.51 build 13313 [ci skip]

sKudum committed:
fix: (Platform) Long title example added for object list item (#4079)

* long title exmample added

* rebase with master

fundamental-bot committed:
chore(release): version 0.25.0-rc.52 build 13324 [ci skip]

mikerodonnell89 committed:
test: (Core) remove test force (#4064)

* remove force

* downport focus trap

* fix failing test

Co-authored-by: InnaAtanasova <i.atanasova@sap.com>

fundamental-bot committed:
chore(release): version 0.25.0-rc.53 build 13334 [ci skip]

dependabot-preview[bot] committed:
chore(deps): bump @stackblitz/sdk from 1.5.1 to 1.5.2 (#4099)

Bumps @stackblitz/sdk from 1.5.1 to 1.5.2.

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

fundamental-bot committed:
chore(release): version 0.25.0-rc.54 build 13344 [ci skip]

dependabot-preview[bot] committed:
chore(deps-dev): bump @storybook/addon-storysource from 6.1.2 to 6.1.11 (#4097)

Bumps [@storybook/addon-storysource](https://github.com/storybookjs/storybook/tree/HEAD/addons/storysource) from 6.1.2 to 6.1.11.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.1.11/addons/storysource)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

fundamental-bot committed:
chore(release): version 0.25.0-rc.55 build 13345 [ci skip]

dependabot-preview[bot] committed:
chore(deps-dev): bump @storybook/addon-links from 6.1.10 to 6.1.11 (#4096)

Bumps [@storybook/addon-links](https://github.com/storybookjs/storybook/tree/HEAD/addons/links) from 6.1.10 to 6.1.11.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.1.11/addons/links)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>
Co-authored-by: deno <mladen.droshev@sap.com>

fundamental-bot committed:
chore(release): version 0.25.0-rc.56 build 13347 [ci skip]

Betrozov committed:
fix: (platform) fix upload collection (#4063)

Betrozov committed:
fix: (platform) fix menu (#4089)

fundamental-bot committed:
chore(release): version 0.25.0-rc.57 build 13353 [ci skip]

JKMarkowski committed:
fix: (Core) Fix structure on Shellbar (#4085)

fundamental-bot committed:
chore(release): version 0.25.0-rc.58 build 13357 [ci skip]

JKMarkowski committed:
fix: (Core) Button emphasized text (#4092)

fundamental-bot committed:
chore(release): version 0.25.0-rc.59 build 13359 [ci skip]

fkolar committed:
fix(platform): Adds missing exports (#4091)

Co-authored-by: fundamental-bot <fundamental@sap.com>

fundamental-bot committed:
chore(release): version 0.25.0-rc.60 build 13360 [ci skip]

JKMarkowski committed:
chore: Increment styles version to 0.13.1 (#4101)

fundamental-bot committed:
chore(release): version 0.25.0-rc.61 build 13361 [ci skip]

katekozlowska committed:
fix: (Core) add fix for Tabs programmatic selection not being applied properly (#4055)

* initial commit

* fix example

fundamental-bot committed:
chore(release): version 0.25.0-rc.62 build 13365 [ci skip]

Lokanathan-k committed:
fix: (Core) address accessibility issues for the Object marker component (#4081)

* fixing the Accessibility issue for the Object marker component

fixing the Accessibility issue for the Object marker component

fundamental-bot committed:
chore(release): version 0.25.0-rc.63 build 13369 [ci skip]

Lokanathan-k committed:
updating the documentation  for the file uplaoder component (#4110)

updating the documentation  for the file Uploader component

fundamental-bot committed:
chore(release): version 0.25.0-rc.64 build 13396 [ci skip]

sKudum committed:
fix: (Platform) list screen reader and accessibility issues. (#4106)

* a11y part1 fix

* a11y part1 fix group header

addressing yuri and inna comments

fundamental-bot committed:
chore(release): version 0.25.0-rc.65 build 13406 [ci skip]

katekozlowska committed:
fix: (Core) add fix for missing styling of selected Tabs (#4011)

* initial commit

* semantic colors temp fix

* routerLink temp style fix

* routerLink temp style fix update

* underline tab stylefx

fundamental-bot committed:
chore(release): version 0.25.0-rc.66 build 13410 [ci skip]

JKMarkowski committed:
docs: Add versions to dorpdown (#4123)

fundamental-bot committed:
chore(release): version 0.25.0-rc.67 build 13412 [ci skip]

kavya-b committed:
fix: (Platform) add tabChange output event to dynamic page (#4084)

* fix: (Platform) handle tab change event for Dynamic Page

- remove hardcoded styles from Dynamic Page

* fix typo and remove transparent bg from example

fundamental-bot committed:
chore(release): version 0.25.0-rc.68 build 13414 [ci skip]

JKMarkowski committed:
fix: (Core) Action sheet buttons left position (#4114)

fundamental-bot committed:
chore(release): version 0.25.0-rc.69 build 13416 [ci skip]

Lokanathan-k committed:
fix: (Platform) Duplicate selection of the dropDown values (#4008)

* fix multi input issues adding hasbyline

fix multi input issues adding hasbyline and fixed the double selection of options

fundamental-bot committed:
chore(release): version 0.25.0-rc.70 build 13424 [ci skip]

fundamental-bot committed:
chore(release): version 0.25.0 build 13432 [ci skip]

JKMarkowski committed:
chore: Fix release script (#4129)

fundamental-bot committed:
chore(release): version 0.25.1-rc.0 build 13436 [ci skip]

fundamental-bot committed:
chore(release): version 0.25.1 build 13438 [ci skip]

InnaAtanasova committed:
fix: (Core) remove DIALOG_CONFIG and DIALOG_REF and make DialogConfig and DialogRef injectable (#4108)

BREAKING CHANGE:
removed DIALOG_CONFIG and DIALOG_REF and made DialogConfig and DialogRef classes injectable

fundamental-bot committed:
chore(release): version 0.26.0-rc.0 build 13442 [ci skip]

mikerodonnell89 committed:
export wizard module (#4134)

fundamental-bot committed:
chore(release): version 0.26.0-rc.1 build 13449 [ci skip]

nikvalor committed:
- Debugging and fixing tests for saucelabs

artolshansky committed:
fix: (Core) Carousel direction defects (#4080)

fundamental-bot committed:
chore(release): version 0.26.0-rc.2 build 13459 [ci skip]

DeepakSap14 committed:
feat: (Platform) Introduce Date picker component (#4109)

* fix:initial implementation

* initial implementation

tests:formcontrol unittest added [ci skip]

docs:refactor

* fix:added template driven code,removed unused code

* docs:formatting changes

* fix:review implementation

* fix:master rebase [ci skip]

* fix:lint fix,stateType added [ci skip]

* fix:review comments

* initial implementation with date adapter

* documentation completed

* test case fixed

* api file change

* review implementations

* fix:code formatting guidelines

* review comments implementation

* Date Picker name consistency

* remove test focus

JKMarkowski committed:
chore: Migrate master to main (#4100)

* chore: Migrate master to main

* Allow main to trigger deploy scripts

fundamental-bot committed:
chore(release): version 0.26.0-rc.3 build 13480 [ci skip]

JKMarkowski committed:
test: CI main change (#4142)

fundamental-bot committed:
chore(release): version 0.26.0-rc.4 build 13484 [ci skip]

Commit:
test: e2e and unit test setup (#3877)

* test(e2e/unit): setup saucelabs run

* test(e2e/unit): update package-lock

* test(e2e/unit): change job run condition

* test(e2e/unit): run jobs in parallel

* test(e2e): add tests for split menu button component

* test(e2e): add menu component e2e tests and refactoring

* test(e2e): add menu button component e2e tests

* update menu button tests to work on FF

* apply rebase and resolve conflicts

* test(e2e): wdio init

* test(e2e): add ts support for wdio

* test(e2e): resolve IE and Safari loading and timeout issues

* test(e2e): fix browser issues

* add fixed card layout e2e tests

* test(e2e): migrate textarea to wdio

* test(e2e): make textaria run on saucelabs

* test(e2e): switch spec migration

* test(e2e): switchfix test

* test(e2e): fix saucelabs config plus convert link and fixed card layout tests to wdio

* test(e2e): updated panel and info-label specs to wdio, updated global wdio methods

* test(e2e): debug link wdio tests and fix config

* test(e2e): convert fixed-card-layout tests to wdio

* tets(e2e): menu spec migration to wdio

* tests(e2e): migrate split-menu-button spec to wdio

* Rollback wdio.conf.js

* PR changes

* test(e2e): debug fixed-card-layout tests on wdio

* test(e2e): add skips for tests broken by issues

* Update link.e2e-spec.ts

* test(e2e): debug switch e2e tests

* Updated scenarios

* test(e2e): debug info label tests

* test(e2e): update info label variables

* test(e2e): update menu checkbox and checkbox-group components, skip Safari and IE

* PR Changes

* test(e2e): fix wdio file after merge

* test(e2e): fix/skip test cases for Safari, IE

* test(e2e): clean protractor test, update config

* test(e2e): code base prittify

* test(e2e): sync package-lock with package.json

* test(e2e/unit): fix log level unskip core unit tests

* test(unit): remove core unit tests run from Lint and Test

* test(e2e): increase e2e test concurrency

* test(e2e): edit spec reporter

* test(e2e): fix PR according to review

* test(e2e): sync package-lock file

* test(e2e): reduce maxInstances to 10

* test(e2e): skip link due error and fix switch

* test(e2e): skip two unstable tests

* skip failed unit and e2e tests

Co-authored-by: anton <anymac@TonyMac.local>
Co-authored-by: Sean Moore <sean.moore@valor-software.com>
Co-authored-by: nikvalor <nikita-lastovych@lavor-software.com>
Co-authored-by: nikvalor <nikita.lastovych+1@valor-software.com>
Co-authored-by: nikvalor <nikita.lastovych@valor-software.com>
Co-authored-by: Sean Moore <47522152+Seamoo13@users.noreply.github.com>

fundamental-bot committed:
chore(release): version 0.26.0-rc.5 build 13497 [ci skip]

dependabot-preview[bot] committed:
chore(deps-dev): bump @storybook/addon-knobs from 6.1.8 to 6.1.11 (#4147)

Bumps [@storybook/addon-knobs](https://github.com/storybookjs/storybook/tree/HEAD/addons/knobs) from 6.1.8 to 6.1.11.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.1.11/addons/knobs)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

fundamental-bot committed:
chore(release): version 0.26.0-rc.6 build 13499 [ci skip]

dependabot-preview[bot] committed:
chore(deps-dev): bump @types/jasmine from 3.6.0 to 3.6.2 (#4146)

Bumps [@types/jasmine](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jasmine) from 3.6.0 to 3.6.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jasmine)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

fundamental-bot committed:
chore(release): version 0.26.0-rc.7 build 13501 [ci skip]

nikvalor committed:
Merge remote-tracking branch 'origin/main' into migrate-combobox-to-wdio

# Conflicts:
#	e2e/wdio/core/tests/fixed-card-layout.e2e-spec.ts
#	e2e/wdio/driver/wdio.ts
#	e2e/wdio/helper/assertion-helper.ts
#	e2e/wdio/platform/fixtures/appData/checkbox-group-page-content.ts
#	e2e/wdio/platform/pages/base-component.po.ts
#	e2e/wdio/platform/pages/checkbox.po.ts
#	e2e/wdio/platform/pages/info-label.po.ts
#	e2e/wdio/platform/pages/panel.po.ts
#	e2e/wdio/platform/pages/split-menu-button.po.ts
#	e2e/wdio/platform/tests/checkbox.e2e-spec.ts
#	e2e/wdio/platform/tests/info-label.e2e-spec.ts
#	e2e/wdio/platform/tests/link.e2e-spec.ts
#	e2e/wdio/platform/tests/menu-button.e2e-spec.ts
#	e2e/wdio/platform/tests/menu.e2e-spec.ts
#	e2e/wdio/platform/tests/panel.e2e-spec.ts
#	e2e/wdio/platform/tests/split-menu-button.e2e-spec.ts
#	e2e/wdio/platform/tests/switch.e2e-spec.ts
#	e2e/wdio/platform/tests/textarea.e2e-spec.ts
#	karma.conf-ci.js
#	package-lock.json
#	package.json
#	wdio.conf.js

nikvalor committed:
Refactored page objects structure
Resolving merge conflicts

nikvalor committed:
Small changes in assertion-helper.ts and wdio.ts

nikvalor committed:
Updated waitForNotDisplayed method

nikvalor committed:
Remove unused file
